X3: Optional container parsing when it is a part of the sequence and the optional parser subject is a sequence too

In Boost 1.70.0, starting with `example/x3/employee.cpp` and applying the diff:
```
diff --git a/employee.cpp b/test.cpp
index 6257102..cb0fdbb 100644
--- a/employee.cpp
+++ b/test.cpp
@@ -31,7 +31,7 @@ namespace client { namespace ast
     {
         int age;
         std::string forename;
-        std::string surname;
+        boost::optional<std::string> surname;
         double salary;
     };
 
@@ -71,7 +71,7 @@ namespace client
             >> '{'
             >>  int_ >> ','
             >>  quoted_string >> ','
-            >>  quoted_string >> ','
+            >>  -quoted_string >> ','
             >>  double_
             >>  '}'
             ;
@@ -118,7 +118,6 @@ main()
 
             std::cout << "-------------------------\n";
             std::cout << "Parsing succeeded\n";
-            std::cout << "got: " << emp << std::endl;
             std::cout << "\n-------------------------\n";
         }
         else
```
results in compiler errors, including
```
/usr/include/boost/spirit/home/x3/support/traits/move_to.hpp:194:24: error: no matching function for call to ‘move_to(std::remove_reference<const char&>::type, std::__cxx11::basic_string<char>&, boost::mpl::identity<boost::spirit::x3::traits::container_attribute>::type)’
```
In contrast, the following diff does successfully compile:
```
diff --git a/employee.cpp b/test.cpp
index 6257102..79f35a0 100644
--- a/employee.cpp
+++ b/test.cpp
@@ -32,7 +32,7 @@ namespace client { namespace ast
         int age;
         std::string forename;
         std::string surname;
-        double salary;
+        boost::optional<double> salary;
     };
 
     using boost::fusion::operator<<;
@@ -72,7 +72,7 @@ namespace client
             >>  int_ >> ','
             >>  quoted_string >> ','
             >>  quoted_string >> ','
-            >>  double_
+            >>  -double_
             >>  '}'
             ;
 
@@ -118,7 +118,6 @@ main()
 
             std::cout << "-------------------------\n";
             std::cout << "Parsing succeeded\n";
-            std::cout << "got: " << emp << std::endl;
             std::cout << "\n-------------------------\n";
         }
         else
```

### Summary
The `optional` operator unary minus (`-`) works as expected when parsing a single `double_` into a `boost::optional<double>` member variable, but does not work as expected when parsing the compound parser `quoted_string` into a corresponding `boost::optional<std::string>`.

--- a/include/boost/spirit/home/x3/operator/optional.hpp
+++ b/include/boost/spirit/home/x3/operator/optional.hpp
@@ -21,6 +21,7 @@ namespace boost { namespace spirit { namespace x3
     struct optional : proxy<Subject, optional<Subject>>
     {
         typedef proxy<Subject, optional<Subject>> base_type;
+        static bool const is_pass_through_unary = false;
         static bool const handles_container = true;
 
         constexpr optional(Subject const& subject)

--- a/test/x3/sequence.cpp
+++ b/test/x3/sequence.cpp
@@ -383,30 +383,6 @@ main()
     }
 
     // test from spirit mailing list
-    // "Optional operator causes string attribute concatenation"
-    {
-        typedef vector<char, char, int> attr_type;
-        attr_type attr;
-
-        auto node = alnum >> -('[' >> alnum >> '=' >> int_ >> ']');
-
-        BOOST_TEST(test_attr("x[y=123]", node, attr));
-        BOOST_TEST(attr == attr_type('x', 'y', 123));
-    }
-
-    // test from spirit mailing list (variation of above)
-    // "Optional operator causes string attribute concatenation"
-    {
-        typedef vector<std::string, std::string, int> attr_type;
-        attr_type attr;
-
-        auto node = +alnum >> -('[' >> +alnum >> '=' >> int_ >> ']');
-
-        BOOST_TEST(test_attr("xxx[yyy=123]", node, attr));
-        BOOST_TEST(attr == attr_type("xxx", "yyy", 123));
-    }
-
-    // test from spirit mailing list
     // "Error with container within sequence"
     {
         typedef vector<std::string> attr_type;
@@ -430,6 +406,23 @@ main()
         BOOST_TEST(at_c<0>(attr).size() == 2);
         BOOST_TEST(at_c<0>(attr)[0] == 123);
         BOOST_TEST(at_c<0>(attr)[1] == 456);
+    }
+
+    { // non-flat optional
+        vector<int, boost::optional<vector<int, int>>> v;
+        auto const p = int_ >> -(':' >> int_ >> '-' >> int_);
+        BOOST_TEST(test_attr("1:2-3", p, v))
+            && BOOST_TEST(at_c<1>(v)) && BOOST_TEST_EQ(at_c<0>(*at_c<1>(v)), 2);
+    }
+
+    { // optional with container attribute
+        vector<char, boost::optional<std::string>> v;
+        auto const p = char_ >> -(':' >> +char_);
+        BOOST_TEST(test_attr("x", p, v))
+            && BOOST_TEST(!at_c<1>(v));
+        v = {};
+        BOOST_TEST(test_attr("x:abc", p, v))
+            && BOOST_TEST(at_c<1>(v)) && BOOST_TEST(*at_c<1>(v) == "abc");
     }
 
     {

